lib/containers/utils.pm

```diff
@@ -315,4 +315,78 @@ sub reset_container_network_if_needed {
     }
 }
 
+sub test_volumes {
+    my %args = @_;
+    my $runtime = $args{runtime};
+
+    # Create a directory with the sticky-bit set
+    assert_script_run('mkdir -m 1777 test');
+
+    # Create Dockerfile with VOLUME defined
+    assert_script_run(q{echo -e 'FROM busybox\nVOLUME /test' > test/Dockerfile});
+
+    # Build image and load it
+    assert_script_run("$runtime build -t test -f test/Dockerfile test/");
+
+    assert_script_run('touch test/test');
+
+    # Test --volumes-from
+    assert_script_run("$runtime run -d --name test test");
+    assert_script_run("$runtime run --rm --volumes-from test test ls /test | grep test");
+    assert_script_run("$runtime rm -vf test");
+
+    assert_script_run("$runtime run -d --name test -v \$PWD/test:/test test");
+    # NOTE: When https://github.com/containers/podman/issues/19529 is solved we must add a version check here
+    my $ret = script_run("$runtime run --rm --volumes-from test test ls /test/test");
+    record_soft_failure("gh#19529 - Unexpected error with --volumes-from") if ($ret != 0 && $runtime == "podman");
+    assert_script_run("$runtime rm -vf test");
+
+    # Test --volume option with directory (read-only)
+    assert_script_run("! $runtime run --rm --volume \$PWD/test:/test:ro test rm /test/test");
+    assert_script_run("test -f test/test");
+
+    # Equivalent --mount option to above
+    assert_script_run("! $runtime run --rm --mount type=bind,source=\$PWD/test,destination=/test,readonly test rm /test/test");
+    assert_script_run("test -f test/test");
+
+    assert_script_run("rm -f test/test");
+
+    # Test --volume option with directory (read-write)
+    assert_script_run("$runtime run --rm --volume \$PWD/test:/test test touch /test/test");
+    assert_script_run("test -f test/test");
+
+    assert_script_run("rm -f test/test");
+
+    # Equivalent --mount option to above
+    assert_script_run("$runtime run --rm --mount type=bind,source=\$PWD/test,destination=/test test touch /test/test");
+    assert_script_run("test -f test/test");
+
+    assert_script_run("rm -f test");
+
+    # Test volume subcommands
+    # Subcommands supported only by podman: exists export import mount reload unmount
+    assert_script_run("$runtime volume create test");
```

Comment:
I would not only create one folume, but a bunch of volumes. This and the next lines can be inside a for loop where we create up to e.g. 10 volumes and perform the same action on all of them.

Reply:
Done.